sdk/core/core-util/src/createAbortablePromise.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError, AbortSignalLike } from "@azure/abort-controller";
+
+/**
+ * Creates an abortable promise.
+ * @param buildPromise - A function that takes the resolve and reject functions as parameters.
+ * @param options - The options for the abortable promise.
+ * @returns A promise that can be aborted.
+ */
+export function createAbortablePromise<T>(
+  buildPromise: (
+    resolve: (value: T | PromiseLike<T>) => void,
+    reject: (reason?: any) => void
+  ) => void,
+  options?: {
+    /** A function to be called if the promise was aborted */
+    cleanupBeforeAbort?: () => void;
+    /** An abort signal */
+    abortSignal?: AbortSignalLike;
```

Comment:
When would I want to create an AbortablePromise without passing an abortSignal? I would have expected this to be required

Reply:
Discussed offline. Since this will be used mostly in library code and the abort signal would come from options.abortSignal from the operation options, the optionality here has the convenience of not forcing us to fork based on the existence of abortSignal